csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/AppInsights/MyAppInsightsLoggerMiddleware.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace NLP_With_Dispatch_Bot.AppInsights
```

Comment:
We should have consistency across JS and C# for samples. My JS dispatch sample does not include app insights. I spoke to Dave and we are adding separate LUIS and QnA Maker samples with app insights. Unless this is required to demonstrate in this sample, we should remove appinsights from this sample.